exporter/splunkhecexporter/client.go

```diff
@@ -56,6 +56,16 @@ type bufferState struct {
 	library  int
 }
 
+// metricBufferState encapsulates intermediate buffer state when pushing metric data
+type metricBufferState struct {
+	buf      *bytes.Buffer
+	tmpBuf   *bytes.Buffer
+	bufFront *metricIndex
+	bufLen   int
+	resource int
+	library  int
+}
```

Comment:
Looks like most of the functionality is copied from logs? Can we reuse existing code to make it easier to maintain? Should be possible by extracting common parts.

For example `logBufferState` and `metricBufferState` seem identical. why can't we use one struct instead?

Reply:
I removed that duplication, good point.